client/v2/tx/tx.go

```diff
@@ -40,7 +44,19 @@ func GenerateOrBroadcastTxCLI(ctx client.Context, flagSet *pflag.FlagSet, msgs .
 		return dryRun(txf, msgs...)
 	}
 
-	return BroadcastTx(ctx, txf, msgs...)
+	broadcaster, err := getBroadcaster(ctx, flagSet)
+	if err != nil {
+		return err
+	}
+
+	return BroadcastTx(ctx, txf, broadcaster, msgs...)
+}
+
+func getBroadcaster(ctx client.Context, flagSet *pflag.FlagSet) (broadcasttypes.Broadcaster, error) {
```

Comment:
That works, but let's precise in the godoc that this uses by default a cometbft broadcaster.
Let's add `GenerateOrBroadcastTxCLIWithBroadcaster` as well so people don't need to re-implement this themselves.